### PR TITLE
feat: add symmetric key monitoring

### PR DESCRIPTION
**Pull request by robrap**
**Description:**

Adds monitoring around JWT decoding and symmetric keys, to help with the eventual deprecation and removal of the symmetric keys.

**Issue:**

See DEPR: Symmetric JWTs:
https://github.com/openedx/public-engineering/issues/83

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bump if needed
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: edx_rest_framework_extensions/auth/jwt/decoder.py [hunk 1]
```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
...
+
+    Arguments:
+        token (str): JWT to be decoded.
+        jwt_issuer (dict): A dict of JWT issuer related settings, containing the symmetric key.
+        decode_symmetric_token (bool): Whether to decode symmetric tokens or not. Pass False for asymmetric tokens only
+    """
+    # .. custom_attribute_name: jwt_auth_decode_symmetric_token
+    # .. custom_attribute_description: True if symmetric keys will also be used for checking
+    #   the JWT signature, and False if only asymmetric keys will be used.
+    set_custom_attribute('jwt_auth_decode_symmetric_token', decode_symmetric_token)
```
Comment (timmc-edx) [inline edx_rest_framework_extensions/auth/jwt/decoder.py:187]: nit: This reads a little misleadingly to me; I would expect it to mean that the token _was_ decoded using a symmetric key, not that the symmetric key is allowed to be used. (Non-blocking.)

File: edx_rest_framework_extensions/auth/jwt/decoder.py [hunk 2]
```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
...
+    #   verification. Should be same as number of asymmetric public keys. This is
+    #   intended to aid in key rotations; once the average count stabilizes at a
+    #   higher number after adding a public key, it should be safe to change the secret key.
+    set_custom_attribute('jwt_auth_verify_asymmetric_keys_count', len(key_set))
+
+    try:
+        _ = JWS().verify_compact(token, key_set)
+        # .. custom_attribute_name: jwt_auth_asymmetric_verified
+        # .. custom_attribute_description: The count of JWTs successfully verified
+        #   using an asymmetric key.
```
Comment (timmc-edx) [inline edx_rest_framework_extensions/auth/jwt/decoder.py:209]: nit: We'll use this as a count, but for each request it's just a boolean.
```suggestion
        # .. custom_attribute_description: Whether the JWT was successfully verified
        #   using an asymmetric key.
```

File: edx_rest_framework_extensions/auth/jwt/decoder.py [hunk 3]
```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
...
     #   the average count stabilizes at a higher number after adding a public key, it
     #   should be safe to change the secret key.
-    set_custom_attribute('jwt_auth_verify_keys_count', len(key_set))
+    set_custom_attribute('jwt_auth_verify_all_keys_count', len(key_set))
 
     try:
         _ = JWS().verify_compact(token, key_set)
+        # .. custom_attribute_name: jwt_auth_symmetric_verified
+        # .. custom_attribute_description: The count of JWTs successfully verified
+        #   using a symmetric key.
```
Comment (timmc-edx) [inline edx_rest_framework_extensions/auth/jwt/decoder.py:234]: Same comment here re: "count".

File: edx_rest_framework_extensions/auth/jwt/decoder.py [hunk 4]
```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
...
+        # Note: A separate custom attribute is used in case there are different JWTs decoded
+        #   in the same request.
```
Comment (timmc-edx) [inline edx_rest_framework_extensions/auth/jwt/decoder.py:236]: I'm having trouble understanding this comment. What other attribute is this being contrasted to?

**Review by timmc-edx — APPROVED** [review]
A couple nits, but everything else looks fine, so approving in advance to unblock.

File: edx_rest_framework_extensions/auth/jwt/decoder.py (hunk 1 above)
Comment (robrap) [reply edx_rest_framework_extensions/auth/jwt/decoder.py]: Updated to `jwt_auth_check_symmetric_key`.

File: edx_rest_framework_extensions/auth/jwt/decoder.py (hunk 4 above)
Comment (robrap) [reply edx_rest_framework_extensions/auth/jwt/decoder.py]: I tried a longer version. You can let me know if it is more clear.

**Review by timmc-edx — APPROVED** [review]
Looks great! (Remember to update changelog date.)

**Pull request merged**